src/configuration.ts

```diff
@@ -1115,8 +1115,7 @@ export async function initFromStateAndSettings(): Promise<void> {
     // More than one setting may be updated on one settings.json save,
     // so make sure to OR the dirty state when it's calculated by a formula (not a simple TRUE value).
     vscode.workspace.onDidChangeConfiguration(async e => {
-        if (vscode.workspace.workspaceFolders &&
-            e.affectsConfiguration('makefile', vscode.workspace.workspaceFolders[0].uri)) {
+        if (vscode.workspace.workspaceFolders && e.affectsConfiguration('makefile')) {
```

Comment:
What is the significance of removing the `vscode.workspace.workspaceFolders[0].uri` argument?

Reply:
When I wrote that code I wanted to avoid all the below analysis when other files were modified and the code path hit than the .vscode/settings.json of that project. But I overlooked the scenario of user settings (defined in another settings.json) and workspace settings (defined in a totally different file, without even the format "settings.json"). VSCode reads from everywhere and knows to apply the rules of defaults/overrides... From wherever it reads, we should only care that the settings of Makefile Tools are impacted and trust the value we're given when reading. I'm sure this is the root cauuse of many bugs.